shapely/prepared.py

```diff
@@ -20,75 +19,53 @@ class PreparedGeometry(object):
       True
     """
 
-    impl = DefaultImplementation
-
     def __init__(self, context):
         if isinstance(context, PreparedGeometry):
             self.context = context.context
         else:
+            pygeos.prepare(context)
             self.context = context
-        self.__geom__ = lgeos.GEOSPrepare(self.context._geom)
         self.prepared = True
 
-    def __del__(self):
-        if self.__geom__ is not None:
-            try:
-                lgeos.GEOSPreparedGeom_destroy(self.__geom__)
-            except AttributeError:
-                pass  # lgeos might be empty on shutdown.
-
-        self.__geom__ = None
-        self.context = None
-        self.prepared = False
-
-    @property
-    def _geom(self):
-        return self.__geom__
-
-    @delegated
     def contains(self, other):
         """Returns True if the geometry contains the other, else False"""
-        return bool(self.impl['prepared_contains'](self, other))
+        return self.context.contains(other)
 
-    @delegated
     def contains_properly(self, other):
         """Returns True if the geometry properly contains the other, else False"""
-        return bool(self.impl['prepared_contains_properly'](self, other))
+        # TODO temporary hack until pygeos exposes contains properly as predicate function
+        from pygeos import STRtree
+        tree = STRtree([other])
+        idx = tree.query(self.context, predicate="contains_properly")
+        return bool(len(idx))
```

Comment:
I opened https://github.com/pygeos/pygeos/pull/267 to add this to pygeos properly